raiden/transfer/mediated_transfer/mediation_fee.py

```diff
@@ -25,24 +26,24 @@ class Interpolate:  # pylint: disable=too-few-public-methods
     def __init__(self, x_list: Sequence, y_list: Sequence) -> None:
         if any(y - x <= 0 for x, y in zip(x_list, x_list[1:])):
             raise ValueError("x_list must be in strictly ascending order!")
-        self.x_list = x_list
-        self.y_list = y_list
-        intervals = zip(x_list, x_list[1:], y_list, y_list[1:])
+        self.x_list = [Fraction(x) for x in x_list]
```

Comment:
Can you add the types of `x_list` and `y_list` in the args of the function as `Sequence[float]`? It will make reading this function easier as I don't have to jump back to see how it's called and what the type of x has to be.

And then perhaps change this (or the argument `x_list`) to a different name and add its type to `Sequence[Fraction]`. Same for `y_list`.

Reply:
I'll make both a `Sequence[Union[Fraction, int]]`. I don't want any float's in there again!

No need to rename, the new variable is `self.x_list` and `x_list`.